### DISCUSSION TITLE
How to download and retain only X most recent videos from a channel?

### DISCUSSION THREAD

**Discussion by ghost** [jmbannon/ytdl-sub#691] [Q&A]
Hi!

Is there a way to download and delete videos based on video count and not by date? For example instead of downloading the last one month of videos from a channel and deleting anything older, could one download the 5 newest videos and delete anything older?

I can think, potentially, of some logic to do this using "playlist_index" or "playlist-index-reversed". However, the only properties for deleting videos that I'm aware of are "keep_files_after" and "keep_files_before". Perhaps it is possible to take the upload date of video with index (or reverse index) 5 and feed it to the "keep_file_after" property.

Wondering if this would be the right approach to take, or does anyone have any other suggestions?

**Comment by jmbannon** [comment]
This is a frequently requested feature that I will add at some point. For the time being, you can use a combination of date_range and ytdl_options.max_downloads as a workaround

**Reply by ghost** [reply]
Thanks for the reply, good to know this is on the radar. I can only apologize for not currently having the skills to assist in development of this feature.

So to confirm there is currently no way to pass a _{playlist_index}_ to the _date_range.after_ or _output_options.keep_files_after_ functions, or to "pull" the upload date from a video with _playlist_index_ _X_ to be used as a variable?

I've tried to think through the logic of the workaround you mentioned but, based on my limited knowledge, it appears to still only allow for deleting files based on a date. Is that correct?

Again, thanks! Your work on this tool is much appreciated!
